sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -187,19 +192,47 @@ def download(self, bucket_name, files_base_path, files_folder_path,
                 rospy.logerr(f"File download failed ({aws_path}). {exception}")
         return download_succeded
 
+    def fast_upload(self, bucketname, s3dir, filelist, progress_func):
+        transfer_config = s3transfer.TransferConfig(
+            use_threads=True,
+            max_concurrency=self.NUM_WORKERS,
```

Comment:
NUM_WORKERS is being used to set `max_concurrency` but also `max_pool_connections`. What is the difference?

Reply:
max_pool_connections is the clients maximum number of connections
max_concurrency is the transfer managers max number of concurrent transfer operations